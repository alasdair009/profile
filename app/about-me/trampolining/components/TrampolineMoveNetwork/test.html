<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
