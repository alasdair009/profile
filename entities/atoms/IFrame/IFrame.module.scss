@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";
@use "sass:map";

.root {
  aspect-ratio: 16 / 9;
  border: none;
  display: block;
  margin: 0 auto;
  max-width: rem.toRem(map.get(dimensions.$sizes, s512));
  width: 100%;
}
