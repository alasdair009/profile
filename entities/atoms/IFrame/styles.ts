"use client";
import styled from "styled-components";
import { sizes } from "@/entities";

export const Root = styled.iframe`
  aspect-ratio: 16 / 9;
  border: none;
  display: block;
  margin: 0 auto;
  max-width: ${sizes.s512.rem};
  width: 100%;
`;
