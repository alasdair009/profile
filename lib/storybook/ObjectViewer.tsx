import { Source } from "@storybook/blocks";

import type { JSX } from "react";

export type ObjectViewerProps = {
  /**
   * The object to display
   */
  object: object;
  /**
   * The title of the code segment
   */
  title: string;
};

export function ObjectViewer({
  object,
  title,
}: ObjectViewerProps): JSX.Element {
  // Format the object as JSON with indentation
  const objectAsString = JSON.stringify(object, null, 4);

  return (
    <div title={title} data-testid={ObjectViewer.name}>
      <Source code={objectAsString} dark={true} />
    </div>
  );
}
