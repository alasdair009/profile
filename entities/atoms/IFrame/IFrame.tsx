import { Root } from "./styles";
import { HTMLAttributes } from "react";

type IFrameProps = {
  /**
   * URL for the iframe content
   */
  src: string;
} & HTMLAttributes<HTMLIFrameElement>;

/**
 * An HTML iframe for external content
 */
export function IFrame({ ...rest }: IFrameProps) {
  return <Root {...rest} allowFullScreen={true} />;
}
