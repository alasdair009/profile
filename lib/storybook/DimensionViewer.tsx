import { Source } from "@storybook/blocks";
import type { HTMLAttributes } from "react";

import { ObjectViewer } from "./ObjectViewer";
import { colors } from "@/entities";

export type DimensionViewerProps = {
  /**
   * The units to display
   */
  units: Record<string, unknown>;
  /**
   * A simplified object that can be used for the DimensionBar interface
   */
  barUnits: Record<string, string | number>;
  /**
   * The title of the code segment
   */
  title: string;
  /**
   * The suffix to display after the unit
   */
  suffix?: string;
  /**
   * Source example
   */
  sourceExample: string;
  /**
   * The name of the object
   */
  objectName: string;
};

export function DimensionViewer({
  units,
  title,
  sourceExample,
  objectName,
  barUnits,
  suffix = "",
}: DimensionViewerProps): JSX.Element {
  return (
    <div data-testid={DimensionViewer.name}>
      <ObjectViewer object={units} title={objectName} />
      <p>
        To use these values import the <strong>{objectName}</strong> object and
        reference the required key.
      </p>
      <Source code={sourceExample} dark={true} language="css" />
      <DimensionBarWrapper title={title}>
        {Object.keys(barUnits).map((key, index) => {
          return (
            <DimensionBar
              key={`${title}${index}`}
              data-testid={DimensionViewer.name}
              width={`${barUnits[key]}${suffix}`}
            >
              <DimensionBarLabel>
                {barUnits[key]}
                {suffix}
                <br />({key})
              </DimensionBarLabel>
            </DimensionBar>
          );
        })}
      </DimensionBarWrapper>
    </div>
  );
}

const DimensionBarWrapper = ({ children }: HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      display: "flex",
      flexDirection: "column",
      overflow: "scroll",
      width: "100%",
    }}
  >
    {children}
  </div>
);

const DimensionBar = ({
  children,
  width,
}: { width: string } & HTMLAttributes<HTMLDivElement>) => (
  <div
    style={{
      background: colors.greenGrass,
      color: "white",
      flex: 1,
      marginBottom: "8px",
      width,
    }}
  >
    {children}
  </div>
);

const DimensionBarLabel = ({ children }: HTMLAttributes<HTMLSpanElement>) => (
  <span
    style={{
      display: "block",
      marginLeft: "8px",
    }}
  >
    {children}
  </span>
);
