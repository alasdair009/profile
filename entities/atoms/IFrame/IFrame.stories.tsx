import { IFrame } from "./IFrame";
import { Meta, StoryObj } from "@storybook/nextjs";
import { expect, within } from "storybook/test";

const meta: Meta<typeof IFrame> = {
  component: IFrame,
  args: {
    src: "https://www.alasdairmacrae.co.uk",
  },
};
export default meta;

export const Default: StoryObj<typeof IFrame> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const iFrameElement = canvas.getByTestId(IFrame.name);

    await expect(iFrameElement).toBeInTheDocument();
    await expect(iFrameElement).toHaveAttribute("src", args.src);
  },
};
