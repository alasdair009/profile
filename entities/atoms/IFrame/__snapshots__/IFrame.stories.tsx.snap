// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/IFrame Default play-test 1`] = `
<iframe src="https://www.alasdairmacrae.co.uk"
        allowfullscreen
        data-testid="IFrame"
        class="sc-hDcvty jZCWOT"
>
</iframe>
`;
